### PR TITLE
replaced AzCopy version update arm64 to x86_64 package

### PR DESCRIPTION
**Pull request by wonwuakpa-msft**
## Description

Addressed mismatch error: `package architecture (arm64) does not match system (amd64)` in the release pipeline. 
Replaced the app architecture to be use a safer and more machine compatible x86_64 package. SAS token in the pipeline is expired -- migrated to EntraID.

- **Feature / Bug Fix**: Replaced the release pipeline pointing to an AMD to x86_64 package.  Migrated the `UpdateVersion` of the pipeline from using SAS tokens for authorization to EntraID
- **Related Links**:
- [ADO Item](https://msazure.visualstudio.com/One/_backlogs/backlog/XStore-DevExp/Epics?showParents=true&workitem=30083102)

## Type of Change

- [X] Bug fix
- [X] New feature
- [ ] Documentation update required
- [ ] Code quality improvement
- [ ] Other (describe):
## How Has This Been Tested?

- Validated the relevant script worked on an Azure VM 
- Validated the release version metadata [file](https://azcopyvnextrelease.blob.core.windows.net/releasemetadata/latest_version.txt) updates to the latest AzCopy version 

Thank you for your contribution to AzCopy!

### CODE REVIEW

**Comment by dphulkar-msft** [comment]
@wonwuakpa-msft can you please test and valid this Update version stage weather its running fine and updating the expected azcopy version in the latest_version.txt file?

File: release-pipeline.yml [hunk 1]
```diff
@@ -2168,6 +2168,8 @@ stages:
 
             - script: |
                 pip install azure-storage-blob
-                python3 releaseVersionUpdate.py "$(VERSION_CNT_SAS_URL)" "`azcopy --version`"
+                AZCOPY_AUTO_LOGIN_TYPE=AzCLI
+                ACCESS_TOKEN=$(az account get-access-token)
```
Comment (gapra-msft) [inline release-pipeline.yml:2172]: Do we need to pass the Access Token? Doesnt look like its being used in the script.

Comment (wonwuakpa-msft) [reply release-pipeline.yml:2172]: I intended to have it to replace the SAS token, but they have different uses cases. So, yes, it's not needed.

File: release-pipeline.yml [hunk 2]
```diff
@@ -2166,8 +2166,32 @@ stages:
...
+                  pip install azure-storage-blob
+                  AZCOPY_AUTO_LOGIN_TYPE=AzCLI
+                  executable_name=azcopy 
+                  
+                  #Extract the version number 
+                  azcopy_version=$(azcopy --version | awk '{print $3}' | head -n 1)
+                  echo "azcopy version: $azcopy_version"
+                  
+                  #Upload to release version blob file
+                  azcopy --version | awk '{print $3}' | head -n 1 | azcopy copy https://azcopyvnextrelease.blob.core.windows.net/releasemetadata/latest_version.txt \
```
Comment (gapra-msft) [inline release-pipeline.yml:2185]: can you first try this script in your VM? might make it easier to catch the error. 

I think the env var Auto login type isnt being recognized

Comment (wonwuakpa-msft) [reply release-pipeline.yml:2185]: I tried it yesterday with no errors. I'm making some changes and try it again in my VM

File: release-pipeline.yml [hunk 3]
```diff
@@ -2166,8 +2166,31 @@ stages:
...
-              displayName: 'Updating version number'
-              workingDirectory: $(root_dir)
+            - task: AzureCLI@2
+              displayName: 'Updating version number with Azure CLI'
+              inputs:
+                azureSubscription: 'ESRP KeyVault identity'
+                scriptType: bash
+                scriptLocation: inlineScript
+                inlineScript: |
+                  pip install azure-storage-blob
```
[the remark this answers is not in the document]
Comment (wonwuakpa-msft) [reply release-pipeline.yml]: Yes, I can delete the file too @dphulkar-msft if that's OK

**Review by gapra-msft — APPROVED** [review]

**Pull request merged**